packages/sdks/src/helpers/flatten.ts

```diff
@@ -22,3 +22,29 @@ export function flatten<T extends Record<string, any>>(
       : { ...acc, [newPath]: value };
   }, {} as T);
 }
+
+/**
+ * Flatten a nested MongoDB query object into a flat object with dot-separated keys.
+ * $ keys are not flattened and are left as is.
+ */
```

Comment:
Could you give some examples in the comment to explain the transformation

Reply:
+1

Reply:
sure! added, please let me know if we should add anything else